// directus/dist/utils/telemetry.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.collectTelemetry = void 0;
const node_machine_id_1 = require("node-machine-id");
const package_json_1 = require("../../package.json");
const env_1 = __importDefault(require("../env"));
const logger_1 = __importDefault(require("../logger"));
async function collectTelemetry() {
    const axios = (await import('axios')).default;
    if (env_1.default['TELEMETRY'] !== false) {
        try {
            await axios.post('https://telemetry.directus.io/', {
                version: package_json_1.version,
                public_url: env_1.default['PUBLIC_URL'],
                project_id: env_1.default['KEY'],
                machine_id: await (0, node_machine_id_1.machineId)(),
                db_client: env_1.default['DB_CLIENT'],
            });
        }
        catch (err) {
            if (env_1.default['NODE_ENV'] === 'development') {
                logger_1.default.error(err);
            }
        }
    }
}
exports.collectTelemetry = collectTelemetry;
